src/app/models/constraint-models/negation-constraint.ts

```diff
@@ -23,29 +23,29 @@ export class NegationConstraint implements Constraint {
     return 'NegationConstraint';
   }
 
-  toQueryObjectWithSubselection(): Object {
+  toQueryObjectWithSubselection(full?: boolean): Object {
```

Comment:
is this flag (full) needed for negation constraint?

Reply:
yes, because previously when you added concept constraint to exclusion criteria, only the path was saved in the query (if you did this for inclusion the full data for concept was added).  Because of this it was impossible to restore the saved query with excluded concept constraint (TMT-216)